otcextensions/tests/functional/sdk/auto_scaling/v1/test_instance.py

```diff
@@ -62,110 +58,164 @@ def _get_default_sec_group(self):
+                        instance=instance, wait=timeout
+                    )
+                return
+        except exceptions.SDKException as e:
+            _logger.warning(
+                'Got exception during getting as instances %s' % e.message
+            )
 
-    def _wait_for_instance(self, as_group):
+    def _wait_for_instance(self):
         timeout = int(os.environ.get('OS_TEST_TIMEOUT'))
         for count in utils.iterate_timeout(
                 timeout=timeout,
                 message="Timeout waiting for instance"
         ):
             instances = list(self.conn.auto_scaling.instances(
-                group=as_group
+                group=TestInstance.AS_GROUP
             ))
             if len(instances) == self.MAX_INSTANCE_NUMBER and instances[0].id:
                 return self.conn.auto_scaling.wait_for_instance(instances[0])
 
-    def _delete_instance(self, instance):
-        timeout = 2 * int(os.environ.get('OS_TEST_TIMEOUT'))
-        self.conn.auto_scaling.remove_instance(
-            instance=instance,
-            delete_instance=True
-        )
-        self.conn.auto_scaling.wait_for_delete_instance(
-            instance=instance,
-            wait=timeout
-        )
+    def _delete_instance(self):
+        if TestInstance.AS_INSTANCE:
+            self.conn.auto_scaling.remove_instance(
+                instance=TestInstance.AS_INSTANCE,
+                delete_instance=True
+            )
+            self.conn.auto_scaling.wait_for_delete_instance(
+                instance=TestInstance.AS_INSTANCE,
+                wait=self.TIMEOUT
+            )
 
     def _initialize_as_group_with_instance(self):
-        self.as_config = self._create_as_config(
-            self._get_image_id(), self._get_default_sec_group()
-        )
-        self.as_group = self._create_as_group(
-            self.as_config.id, self.infra.get("router_id"),
-            self.infra.get("network_id")
-        )
-        self.as_instance = self._wait_for_instance(
-            self.as_group
-        )
+        self._create_as_config()
+        self._create_as_group()
+        TestInstance.AS_INSTANCE = self._wait_for_instance()
 
     def _deinitialize_as_group_with_instance(self):
-        if self.as_instance:
-            self._delete_instance(self.as_instance)
-        if self.as_group:
-            self._delete_as_group(self.as_group)
-        if self.as_config:
-            self._delete_as_config(self.as_config)
-
-    def test_find_instance_by_id(self):
+        self._delete_instance()
+        self._delete_as_group()
+        self._delete_as_config()
+
+    def test_01_create_as_group(self):
+        as_group = self.conn.auto_scaling.find_group(
+            name_or_id=self.AS_GROUP_NAME)
+        self.assertIsNotNone(as_group)
+        self.assertEqual(as_group.name, self.AS_GROUP_NAME)
+
+    def test_02_simple_delete_as_group(self):
+        self._delete_as_group()
```

Comment:
i think we do not need this test, because deleting chack in _deinitialize... method